Makefile

```diff
@@ -41,9 +41,18 @@
 #   make -f tools/Makefile.win
 #
 
--include .config
+ifeq ($(wildcard .config),)
```

Comment:
@xiaoxiang781216 we will relay only on configure.sh (and the configuration step) to check if Make.defs exists?  (I see that you removed the check_context target)

Reply:
No, the check still inside make, but the location change from tools/Makefile.unix or tools/Makefile.win to Makefile:
```
ifeq ($(wildcard .config),)
.DEFAULT:
	@echo "Nuttx has not been configured:"
	@echo "  tools/configure.sh <target>"

clean distclean:
	@:
else
include .config
ifeq ($(CONFIG_WINDOWS_NATIVE),y)
include tools/Makefile.win
else
include tools/Makefile.unix
endif
endif
```
If .config doesn't exist, .DEFAULT will output the same message as check_context.

Reply:
Yes, but that only checks for `.config` not `Make.defs`

Reply:
is it enough to only check .config? I can add more code to check Make.defs but which make Makefile a little bit hard to read. Make.defs and .config appear at the same in most case. If something go to wrong, include Make.defs will stop immediately and report the problem cleanly.

Reply:
I believe .config is enough since they are both copied at about the same time.  I was just trying to see if this is what you had in mind and that I understood the change correctly.